h2o-r/h2o-package/R/kvstore.R

```diff
@@ -76,6 +76,14 @@ h2o.removeAll <- function(timeout_secs=0, retained_elements = c()) {
         retained_keys <- append(retained_keys, element@model_id)
       } else if (is.H2OFrame(element)) {
         retained_keys <- append(retained_keys, h2o.getId(element))
+      } else if( is.character(element) ) {
+        if (is.element(element, h2o.ls()$key)) {
```

Comment:
I think we should skip this validation and let the backend handle the error.

If the user is providing a key, they probably know what they are doing. Moreover, this still doesn't cover all cases (key can be valid but it isn't a model/frame).

I would rather keep this simple.

Reply:
This validation is indeed done on backend. In `DKVManager`, roughly on line 53:
```java
        throw new IllegalArgumentException(String.format("Given key %s is of type %d. Please provide only Model and Frame keys.", key.toString(), value.type()));
```

There are no clear rules for how much validation we should use. However, I managed to find one more argument for the approach @michalkurka proposes - only backend knows whether a key is a proper key or nonsense. We even have structures like `KeyV3.java`. 

Any exceptions on the backend are "nicely" propagated to the user anyway, so we're kind of safe here.

Reply:
I removed the check.

My idea was more simple and reflect my experience with removeAll function, when I accidentally pass wrong key and remove all instead of saying "Hey user, you want to retain something I don't know, are you sure?". But it is true that if we cannot do it properly then this "half-way frontend" check could cause hard to reproduce bugs.

Reply:
> My idea was more simple and reflect my experience with removeAll function, when I accidentally pass wrong key and remove all instead of saying "Hey user, you want to retain something I don't know, are you sure?". But it is true that if we cannot do it properly then this "half-way frontend" check could cause hard to reproduce bugs.

@valenad1 Good thinking, I appreciate you trying to go the extra mile.

We also need to keep in mind that the H2O cluster can easily have tens of thousands of models (eg. in the case of Segment Models). The calling h2o.ls for every single retained key gets expensive.

Generally: my guidance is - validate only using the information you already have. If wouldn't need to ask the backend - let the backend do the validation instead.